crates/rpc/src/client/interceptor.rs

```diff
@@ -0,0 +1,30 @@
+use std::collections::HashMap;
+
+use tonic::{metadata::AsciiMetadataValue, service::Interceptor};
+
+/// Interceptor designed to inject required metadata into all [`super::ApiClient`] requests.
+#[derive(Default, Clone)]
+pub struct MetadataInterceptor {
+    metadata: HashMap<&'static str, AsciiMetadataValue>,
+}
+
+impl MetadataInterceptor {
+    /// Adds or overwrites HTTP ACCEPT metadata to the interceptor.
+    ///
+    /// Provided version string must be ASCII.
+    pub fn with_accept_metadata(mut self, version: &str) -> Result<Self, anyhow::Error> {
+        let accept_value = format!("application/vnd.miden.{version}+grpc");
```

Comment:
Not for this PR, but there should be no reason why we could not in the future generalize `{version}` to be the commit hash of the `miden-base` dependency to solve [this problem](https://github.com/0xMiden/miden-node/issues/804#issuecomment-2832401324:~:text=making%20it%20required.-,There%20is%20also%20a%20different%20but%20related%20problem%20that%20we%20can,wonder%20if%20we%20could%20do%20something%20for%20this%20case%20as%20well.,-igamigo), right?

Reply:
I think that is more of a prerequisite of our semver-based releases rather than something that would need to be enforced here. As in compatible node/client releases based on semver must have compatible base dependancies.

I may have misunderstood.